serveradmin/api/decorators.py

```diff
@@ -77,20 +76,110 @@ def _wrapper(request):
     return update_wrapper(_wrapper, view)
 
 
-def authenticate_app(app, token, timestamp, now, body):
+def authenticate_app(
+    public_keys, signatures, app_id, token, timestamp, now, body
+):
+    """Authenticate requests
+
+    Ensure this request isn't beeing replayed by making sure the timestamp
+    contained in the request is no older then 300 seconds or raise
+    PermissionDenied.
+
+    Hand over the real verification of auth token HMACs to authenticate_app_psk
+    or public key signatures to authenticate_app_ssh.
+
+    Ensure that the application and applications owner aren't deactivated or
+    raise PermissionDenied.
+
+    Return the app the user authenticated to
+    """
     if timestamp + 300 < now:
```

Comment:
Maybe even make if configurable via server settings.

Reply:
I can make it a constant, set it lower, whatever you want. I don't think it really matters as only a nonce would the way to really prevent replaying.

Let's not make this configurable though. There is no use case on earth where it would make sense to change this. Such settings are just spam to potential users.

Reply:
Good to go for me. I would go for 16 seconds. But maybe Mr. Security has some number, too.

Reply:
Do you think he has a d16? :trollface:

Reply:
I heard drums and a snake falling down a cliff here.

Reply:
maybe worth another pull request as well. at some point. i mean implementing a nonce :)

Reply:
Changed to +-16s